### PR TITLE
subsys: bluetooth: controller: Connection AoA TX

### PR DESCRIPTION
**Pull request by Tschet1**
Add support for the Connection AoA transmitter feature.

Signed-off-by: Jan Müller <jan.mueller@nordicsemi.no>

### CODE REVIEW

File: subsys/bluetooth/controller/Kconfig [hunk 1]
```diff
@@ -176,10 +176,10 @@ config BT_CTLR_SDC_RX_STACK_SIZE
 # CONFIG_BT_CTLR_DF is declared in Zephyr and also here for a second time,
 # to avoid BT_CTLR_DF_SUPPORT dependency.
 config BT_CTLR_DF
```
Comment (Tschet1) [inline subsys/bluetooth/controller/Kconfig:178]: @ryanjh : Any suggestion as to how we should make the new connection AoA transmitter feature experimental?

One way would be something like

`
config BT_CTLR_DF
  bool

config BT_CTLR_SDC_CONNECTION_AOA
  bool "blabla"
  select EXPERIMENTAL
  select BT_CTLR_DF
  select BT_CTLR_DF_CONN_CTE_TX

config BT_CTLR_SDC_CONNECIONLESS_AOA
  bool "blabla!
  select BT_CTLR_DF
  select BT_CTLR_DF_ADV_CTE_TX
`

Any opinion on this?

File: subsys/bluetooth/controller/hci_internal.c [hunk 2]
```diff
@@ -358,6 +358,14 @@ static void supported_commands(sdc_hci_ip_supported_commands_t *cmds)
 
 	cmds->hci_le_set_connectionless_cte_transmit_parameters = 1;
 	cmds->hci_le_set_connectionless_cte_transmit_enable = 1;
+#endif
+
+#if defined(CONFIG_BT_CTLR_DF_CONN_CTE_RSP)
+	cmds->hci_le_connection_cte_response_enable = 1;
```
Comment (ryanjh) [inline subsys/bluetooth/controller/hci_internal.c:364]: cmds->hci_le_transmitter_test_v3 is required by both CONFIG_BT_CTLR_DF_ADV_CTE_TX and CONFIG_BT_CTLR_DF_CONN_CTE_RSP.

File: subsys/bluetooth/controller/Kconfig (hunk 1 above)
Comment (Tschet1) [reply subsys/bluetooth/controller/Kconfig:178]: agreed as not needed

**Review by ryanjh — APPROVED** [review]

File: doc/nrf/releases/release-notes-changelog.rst [hunk 3]
```diff
@@ -54,6 +54,7 @@ Bluetooth LE
 * Added support for changing the radio transmitter's default power level using the :c:func:`sdc_default_tx_power_set` function.
 * Added support for changing the peripheral latency mode using the :c:func:`sdc_hci_cmd_vs_peripheral_latency_mode_set` function.
 * Added support for changing the default TX power using ``CONFIG_BT_CTLR_TX_PWR_*``.
+* Added support for the Connection CTE Response feature in the Angular-of-Arrival configuration to the SoftDevice controller.
```
Comment (yuxCai) [inline doc/nrf/releases/release-notes-changelog.rst:57]: should it be angle of arrival instead?

**Review by rugeGerritsen — APPROVED** [review]

**Pull request merged**